### PR TITLE
[4.0] Removing special layouts for com_contact

### PR DESCRIPTION
**Pull request by Hackwar**
The special layouts of com_contact (displaying contacts in sliders, tabs or plain) are stupid. We have no other output that you can switch like this. So this PR removes that feature.

### How to test
Create a new contact and see that you can't select a presentation style for this. Likewise in menu items and the global configuration for com_contact. See in the frontend that a contact is only displayed in plain style.

### CODE REVIEW

**Comment by ciar4n** [comment]
Related issue.. https://github.com/joomla/joomla-cms/issues/15345

**Comment by infograf768** [comment]
Qualifying this feature as stupid is your analysis.
Let's each one select what they want as it is already available.

**Comment by C-Lodder** [comment]
Fully agree with this.

Easier to maintain and users can create an override, should they wish to change the structure

**Comment by mbabker** [comment]
> Let's each one select what they want as it is already available.

Then re-implement it as alternative layouts instead of shoving what is in essence three different layout styles into one template (basically doing the same as default and blog category layouts for articles).  I don't think the feature is "stupid" necessarily but the implementation just sucks and makes building overrides more complex than necessary because you have to decipher what code belongs to what layout option and account for a lot of scenarios that might not exist if you're going for the plain style.

**Comment by brianteeman** [comment]
If you are going to remove the options then you also should remove the language strings

**Comment by Hackwar** [comment]
Excuse my language. Displaying the contact in sliders or tabs is of course not stupid, but as @mbabker wrote, the implementation is among the worst that I can think of.

@brianteeman going to remove the language strings now.

File: components/com_contact/tmpl/contact/default.php [hunk 1]
```diff
@@ -110,143 +96,36 @@
...
 
 		<?php echo $this->loadTemplate('articles'); ?>
-
-		<?php if ($presentation_style === 'sliders') : ?>
-			<?php echo HTMLHelper::_('bootstrap.endSlide'); ?>
-		<?php elseif ($presentation_style === 'tabs') : ?>
-			<?php echo HTMLHelper::_('uitab.endTab'); ?>
-		<?php endif; ?>
-	<?php endif; ?>
+    <?php endif; ?>
```
Comment (Quy) [inline components/com_contact/tmpl/contact/default.php]:
```suggestion
	<?php endif; ?>
```

File: components/com_contact/tmpl/contact/default.php [hunk 2]
```diff
@@ -110,143 +96,36 @@
...
 
 		<?php echo $this->loadTemplate('profile'); ?>
-
-		<?php if ($presentation_style === 'sliders') : ?>
-			<?php echo HTMLHelper::_('bootstrap.endSlide'); ?>
-		<?php elseif ($presentation_style === 'tabs') : ?>
-			<?php echo HTMLHelper::_('uitab.endTab'); ?>
-		<?php endif; ?>
-	<?php endif; ?>
+    <?php endif; ?>
```
Comment (Quy) [inline components/com_contact/tmpl/contact/default.php]:
```suggestion
	<?php endif; ?>
```

File: components/com_contact/tmpl/contact/default.php [hunk 3]
```diff
@@ -262,19 +141,6 @@
...
-		<?php endif; ?>
-	<?php endif; ?>
-
-	<?php if ($accordionStarted) : ?>
-		<?php echo HTMLHelper::_('bootstrap.endAccordion'); ?>
-	<?php elseif ($tabSetStarted) : ?>
-		<?php echo HTMLHelper::_('uitab.endTabSet'); ?>
-	<?php endif; ?>
-
+    <?php endif; ?>
```
Comment (Quy) [inline components/com_contact/tmpl/contact/default.php]:
```suggestion
	<?php endif; ?>
```

File: components/com_contact/tmpl/contact/default_user_custom_fields.php [hunk 4]
```diff
@@ -36,13 +35,7 @@
...
 <?php foreach ($userFieldGroups as $groupTitle => $fields) : ?>
 	<?php $id = ApplicationHelper::stringURLSafe($groupTitle); ?>
-	<?php if ($presentation_style == 'sliders') : ?>
-		<?php echo HTMLHelper::_('bootstrap.addSlide', 'slide-contact', $groupTitle ?: Text::_('COM_CONTACT_USER_FIELDS'), 'display-' . $id); ?>
-	<?php elseif ($presentation_style == 'tabs') : ?>
-		<?php echo HTMLHelper::_('bootstrap.addTab', 'myTab', 'display-' . $id, $groupTitle ?: Text::_('COM_CONTACT_USER_FIELDS')); ?>
-	<?php elseif ($presentation_style == 'plain') : ?>
-		<?php echo '<h3>' . ($groupTitle ?: Text::_('COM_CONTACT_USER_FIELDS')) . '</h3>'; ?>
-	<?php endif; ?>
+    <?php echo '<h3>' . ($groupTitle ?: Text::_('COM_CONTACT_USER_FIELDS')) . '</h3>'; ?>
```
Comment (Quy) [inline components/com_contact/tmpl/contact/default_user_custom_fields.php]:
```suggestion
	<?php echo '<h3>' . ($groupTitle ?: Text::_('COM_CONTACT_USER_FIELDS')) . '</h3>'; ?>
```

**Comment by astridx** [comment]
I tested this PR and I like it. In my opinion, the presentation belongs  only to the template. 

But: Does not the parameter have to be deleted here

`https://github.com/joomla/joomla-cms/blob/ad2c673ac27c2fd19bbdf6003aee0451d068de16/installation/sql/mysql/joomla.sql#L521`

and here

`https://github.com/joomla/joomla-cms/blob/ad2c673ac27c2fd19bbdf6003aee0451d068de16/installation/sql/postgresql/joomla.sql#L531`

**Comment by Hackwar** [comment]
Hello @astridx, thanks for your test. Yes, the parameter data is still there in the SQL, but it will simply be ignored. When the data is loaded, that key-value-pair will simply not be added anywhere and if you save the global config of com_contact the next time, this would be gone. I've not deleted these entries because I'm a bit lazy and I expect us to at one time to update all these parameters from the XML source anyway before 4.0 is released.

**Comment by Hackwar** [comment]
Thank you!

**Pull request merged**